.github/workflows/gradle.yml

```diff
@@ -0,0 +1,36 @@
+name: Test Guides
+on:
+  push:
+    branches:
+      - master
+      - '[1-9]+.[0-9]+.x'
+  pull_request:
+    branches:
+      - master
+      - '[1-9]+.[0-9]+.x'
+jobs:
+  alltests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: ["8","11","17"]
+    env:
+      JDK_VERSION: ${{ matrix.java }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Run All Guide Tests
+        run: 'export AWS_ACCESS_KEY_ID=XXX;export AWS_SECRET_ACCESS_KEY=YYY;export AWS_REGION=us-east-1;./gradlew runAllGuideTests'
```

Comment:
Should these `export` lines be in an `env` block?  And not sure XXX and YYY is correct unless there's some magic that replaces them?

Reply:
Moved them